tribits/ci_support/CreateIssueTrackerFromCDashQuery.py

```diff
@@ -0,0 +1,271 @@
+
+
+  def __init__(self, issueTrackerFormatter, cdashProjectStartTimeUtc=None,
+      usageHelp="", issueTrackerUrlTemplate="", issueTrackerTemplate="",
+    ):
+    self.issueTrackerFormatter = issueTrackerFormatter
+    self.cdashProjectStartTimeUtc = cdashProjectStartTimeUtc
+    self.usageHelp = usageHelp
+    self.issueTrackerUrlTemplate = issueTrackerUrlTemplate
+    self.issueTrackerTemplate = issueTrackerTemplate
+    self.options = None
+
+
+  def runDriver(self):
+
+    print("\n***")
+    print("*** Getting data to create a new issue tracker")
+    print("***\n")
+
+    self.getCmndLineOptions()
+
+    print("Downloading full list of nonpassing tests from CDash URL:\n")
+    print("   "+self.options.nonpassingTestsUrl+"\n")
+
+    nonpassingTestsLOD = self.downloadNonpassingTestsData()
+    print("\nTotal number of nonpassing tests over all days = "\
+      +str(len(nonpassingTestsLOD)))
+
+    uniqNonpassingTestsLOD = getUniqueTestsListOfDicts(nonpassingTestsLOD)
+    print("\nTotal number of unique nonpassing test/build pairs over all days = "\
+          +str(len(uniqNonpassingTestsLOD)))
+
+    (testnameList, buildnameList) = getTestnameAndBuildnameLists(uniqNonpassingTestsLOD)
+    print("\nNumber of test names = "+str(len(testnameList)))
+    print("\nNumber of build names = "+str(len(buildnameList)))
+
+    testingDayStartNonpassingDate = getTestingDayStartNonpassingDate(
+      nonpassingTestsLOD, self.options.cdashProjectStartTimeUtc)
+    #print("testingDayStartNonpassingDate = "+testingDayStartNonpassingDate)
+
+    testHistoryHtmlTableText = ""  # ToDo: Implement!
+
+    issueTrackerText = self.issueTrackerFormatter.createFormattedIssueTracker(
+      IssueTrackerData(
+        summaryLine=self.options.summaryLine,
+        testingDayStartNonpassingDate=testingDayStartNonpassingDate,
+        nonpassingTestsUrl=self.options.nonpassingTestsUrl,
+        uniqNonpassingTestsLOD=uniqNonpassingTestsLOD,
+        buildnameList=buildnameList,
+        testnameList=testnameList,
+        testHistoryHtmlTableText=testHistoryHtmlTableText,
+        )
+      )
+
+    self.writeNewIssueTrackerFile(issueTrackerText)
+
+    self.writeNewTestsWithIssueTrackersFile(uniqNonpassingTestsLOD)
+
+
+  def injectExtraCmndLineOptions(self, clp):
+    clp.add_argument(
+      "--nonpassing-tests-url", "-u", dest="nonpassingTestsUrl", default="",
+      help="Full CDash queryTest.php URL for the list of nonpassing tests over a"\
+        +" time range given as 'begin' and 'end' fields [Required]." )
+    clp.add_argument(
+      "--cdash-project-start-time", dest="cdashProjectStartTimeUtc", default="",
+      help="Starting time for the CDash testing day in 'hh:mm' in UTC."\
+        + " Check the CDash project settings for the testing day start time." )
+    clp.add_argument(
+      "--summary-line", "-s", dest="summaryLine", default="<summary-line>",
+      help="The summary line text for the issue tracker [Default '<summary-line>']." )
```

Comment:
I don't think you need to specify the default value in the `help` if it's already specified in the `default`.

Reply:
> I don't think you need to specify the default value in the help if it's already specified in the default.

That does not seem to be the case.  I just ran --help for the example program and it produced:

```
  --summary-line SUMMARYLINE, -s SUMMARYLINE
                        The summary line text for the issue tracker [Default
                        '<summary-line>'].
```

Reply:
Oh, I wonder if it differs by Python version.  Never mind then.

Reply:
@jmgate, even with Python 3.5.2 I still get:

```
  --summary-line SUMMARYLINE, -s SUMMARYLINE
                        The summary line text for the issue tracker [Default
                        '<summary-line>'].
 ```

Reply:
Ah, [this is what it was](https://stackoverflow.com/a/12151325).

Reply:
Okay, nice to know about `formatter_class=argparse.ArgumentDefaultsHelpFormatter`.